### PR TITLE
Clarify return value of the sequence lookup

### LINKED ISSUE #17852: with_sequence should provide int (not unicode) values
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

with_sequence
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file =
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

No changes
##### OS / ENVIRONMENT

N/A
##### SUMMARY

`with_sequence` sets item to a series of unicode values instead of integer values. This makes it impossible to do arithmetic with them.
##### STEPS TO REPRODUCE

This fails:

```
- hosts: localhost
  tasks:
    - debug: msg="hello {{10+item}}"
      with_sequence: start=1 end=9
```

This works:

```
- hosts: localhost
  tasks:
    - debug: msg="hello {{10+item}}"
      with_items: [1, 2, 3, 4, 5, 6, 7, 8, 9]
```

##### EXPECTED RESULTS

"hello 11" to "hello 19"
##### ACTUAL RESULTS

```
TASK [debug] *******************************************************************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "Unexpected templating type error occurred on (hello {{10+item}}): unsupported operand type(s) for +: 'int' and 'unicode'"}
```

But using `with_items` instead of `with_sequence` it's fine:

```
ok: [localhost] => (item=1) => {
    "item": 1,
    "msg": "hello 11"
}
ok: [localhost] => (item=2) => {
    "item": 2,
    "msg": "hello 12"
}
... etc
```

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes #17852

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

sequence lookup

### CODE REVIEW

File: lib/ansible/plugins/lookup/sequence.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
       - 'Arguments can be specified as key=value pair strings or as a shortcut form of the arguments string is also accepted: [start-]end[/stride][:format].'
       - 'Numerical values can be specified in decimal, hexadecimal (0x3f8) or octal (0600).'
       - Starting at version 1.9.2, negative strides are allowed.
+      - Generated items are strings. Use Jinja2 filters to convert items to preferred type, e.g. ``{{ 1 + item|int }}``.
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/sequence.py:18]: This is a bit incorrect, like for all lookups, generated items might be a specific type (in this case, `ints`) but by default JInja2 converts everything into strings so you either enabled 'native types' or use casting on consumption.

This should be a general warning in the main lookup plugin page

Comment (mkrizek) [reply lib/ansible/plugins/lookup/sequence.py:18]: https://github.com/ansible/ansible/blob/222adde00e0ac2e73addd23be77c5afbd7ae6291/lib/ansible/plugins/lookup/sequence.py#L228

Comment (mkrizek) [reply lib/ansible/plugins/lookup/sequence.py:18]: No, this is sequence lookup specific. All items are formatted (even if you don't specify the format) and returned as strings. Jinja2 comes into play later and fails if you try to do unsupported operation on strings, native types won't help with this.

Comment (bcoca) [reply lib/ansible/plugins/lookup/sequence.py:18]: ah, true, i forgot the `format` option forces always to stringify

Comment (bcoca) [reply lib/ansible/plugins/lookup/sequence.py:18]: I would also point out that the `range` filter is a better alternative

**Pull request merged**